app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,79 @@
+<section class="m-3 mx-5">
+  <h1 class="text-center"><%= @recipe.name %></h1>
+  <div class="d-flex justify-content-between">
+    <p>Preparation time: <%= @recipe.preparation_time %>
+      <% if @recipe.preparation_time <= 1 %>
+        hour
+      <% else %>
+        hours
+      <% end %>
+    </p>
+    <% if can? :read, @recipe %>
+      <div class="d-flex align-items-center">
+        <span class="me-1">
+          <% if @recipe.public %>
+            Public
+          <% else %>
+            Private
+          <% end %>
+        </span>
+        <%= form_with(model: @recipe, local: true, html: { method: :patch, id: 'recipe-form' }) do |form| %>
+          <div class="form-check form-switch">
+            <%= form.check_box :public, class: 'form-check-input', id: 'publicSwitch' %>
+          </div>
+        <% end %>
+      </div>
+    <% end %>
+  </div>
+  <p>Cooking time: <%= @recipe.cooking_time %>
+    <% if @recipe.preparation_time <= 1 %>
+      hour
+    <% else %>
+      hours
+    <% end %>
+  </p>
+  <% if (can? :read, @recipe) || @recipe.public %>
+    <p><%= @recipe.description %></p>
+    <div class="d-flex justify-content-between mt-3">
+      <button type="button" class="btn btn-primary" data-bs-toggle="modal" data-bs-target="#exampleModal">
+        Generate shopping list
+      </button>
+      <% if can? :read, @recipe %>
+        <%= link_to 'Add ingredient', new_recipe_recipes_food_path(@recipe), class: "btn btn-primary" %>
+      <% end %>
+    </div>
+    <table class="table mt-3">
+      <thead>
+        <tr>
+          <th scope="col">Food</th>
+          <th scope="col">Quantity</th>
+          <th scope="col">Value</th>
+          <th scope="col">Actions</th>
+        </tr>
+      </thead>
+      <tbody>
+        <%= render partial: 'recipe_food', collection: @recipe_foods %>
+      </tbody>
+    </table>
+  <% else %>
+    <p class="text-center fs-3">This is private a recipe.</p>
+  <% end %>
+</section>
+<div class="modal fade" id="exampleModal" tabindex="-1" aria-labelledby="exampleModalLabel" aria-hidden="true">
+  <div class="modal-dialog">
+    <div class="modal-content">
+      <div class="modal-header">
+        <h1 class="modal-title fs-5" id="exampleModalLabel">Generating Shopping List</h1>
+        <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
+      </div>
+      <div class="modal-body">
+        <h2>Choose an inventory to use</h2>
+        <%= form_with url: shopping_list_path, method: :get do |form|%>
+          <%= form.select :inventory_id, options_from_collection_for_select(Inventory.all, 'id', 'name'), { prompt: 'Select an inventory' }, { class: 'form-control', required: true } %>
+          <%= form.submit 'Generate', class: 'btn btn-primary' %>
+          <%= form.hidden_field :recipe_id, value: @recipe.id %>
+        <% end %>
+      </div>
+    </div>
+  </div>
+</div>
```

Comment:
- [ ] Great job on implementing the modal for selecting an inventory to create a shopping list! :+1: However, there seems to be a layer covering the entire browser when clicking on the **Generate shopping list** button, making it impossible to click on any buttons within the modal. Please investigate and address this issue.

Reply:
@LionRouge1 thanks for noticing this. this was being caused by a rogue css class. I have fixed the issue :)